released/discovery_center/mission_4033/main.tf

```diff
@@ -0,0 +1,201 @@
+###############################################################################################
+# Setup of names in accordance to naming convention
+###############################################################################################
+resource "random_uuid" "uuid" {}
+
+locals {
+  random_uuid               = random_uuid.uuid.result
+  project_subaccount_domain = lower(replace("mission-4033-${local.random_uuid}", "_", "-"))
+  project_subaccount_cf_org = substr(replace("${local.project_subaccount_domain}", "-", ""), 0, 32)
+}
+
+###############################################################################################
+# Creation of subaccount
+###############################################################################################
+resource "btp_subaccount" "project" {
+  name      = var.subaccount_name
+  subdomain = local.project_subaccount_domain
+  region    = lower(var.region)
+}
+
+###############################################################################################
+# Assignment of users as sub account administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-admins" {
+  for_each             = toset("${var.subaccount_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Administrator"
+  user_name            = each.value
+}
+
+###############################################################################################
+# Assignment of users as sub account service administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-service-admins" {
+  for_each             = toset("${var.subaccount_service_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Service Administrator"
+  user_name            = each.value
+}
+
+######################################################################
+# Add "sleep" resource for generic purposes
+######################################################################
+resource "time_sleep" "wait_a_few_seconds" {
+  create_duration = "30s"
+}
+
+######################################################################
+# Setup Kyma
+######################################################################
+data "btp_regions" "all" {}
+
+locals {
+  subaccount_iaas_provider = [for region in data.btp_regions.all.values : region if region.region == data.btp_subaccount.this.region][0].iaas_provider
+}
+
+data "btp_subaccount" "this" {
```

Comment:
This is confusing for the reader. This data source should be placed before the locals